src/components/main/actionsPanel/workspaceTreeView/helpers/moveActions.ts

```diff
@@ -0,0 +1,226 @@
+import {useContext} from 'react';
+
+import { 
+	TFileNodeData, 
+	createDirectory, 
+	getStat, 
+	readDir, 
+	readFile, 
+	removeFileSystem, 
+	writeFile 
+} from '@_node/file';
+import { MainContext } from '@_redux/main';
+
+
+export const moveActions = () =>{
+
+  const { addMessage } = useContext(MainContext);
+
+	const moveLocalFF = async (
+		handler: FileSystemHandle,
+		parentHandler: FileSystemDirectoryHandle,
+		targetHandler: FileSystemDirectoryHandle,
+		newName: string,
+		copy: boolean = false,
+		showWarning: boolean = false,
+	  ) => {
+		if (handler.kind === "directory") {
+		  // validate if the new name exists
+		  let exists = true;
+		  try {
+			await targetHandler.getDirectoryHandle(newName, { create: false });
+			exists = true;
+		  } catch (err) {
+			exists = false;
+		  }
+		  if (exists) {
+			showWarning &&
+			  addMessage({
+				type: "error",
+				content: "Folder with the same name already exists.",
+			  });
+			return;
+		  }
+	
+		  // move nested handler-dir to targetHandler with the newName - copy (optional)
+		try {
+			const newHandler = await targetHandler.getDirectoryHandle(newName, {
+			  create: true,
+			});
+			const dirQueue = [
+			  {
+				source: handler as FileSystemDirectoryHandle,
+				destination: newHandler,
+			  },
+			];
+		  
+			while (dirQueue.length) {
+			  const { source, destination } = dirQueue.shift() as {
+				source: FileSystemDirectoryHandle;
+				destination: FileSystemDirectoryHandle;
+			  };
+		  
+			  for await (const entry of source.values()) {
+				if (entry.kind === "directory") {
+				  const newDir = await destination.getDirectoryHandle(entry.name, {
+					create: true,
+				  });
+				  dirQueue.push({
+					source: entry as FileSystemDirectoryHandle,
+					destination: newDir,
+				  });
+				} else {
+				  const newFile = await destination.getFileHandle(entry.name, {
+					create: true,
+				  });
+				  const content = await (entry as FileSystemFileHandle).getFile();
+				  const writableStream = await newFile.createWritable();
+				  await writableStream.write(content);
+				  await writableStream.close();
+				}
+			  }
+			}
+		  
+			// Handle copy (optional)
+			!copy && (await parentHandler.removeEntry(handler.name, { recursive: true }));
+		  } catch (err) {
+			throw new Error("error");
+		  }
+		  
+
+		} else {
+		  // validate if the new name exists
+		  let exists = true;
```

Comment:
Try to convert the else block also into a separate function with a meaningful name